huaweicloud/services/dds/resource_huaweicloud_dds_backup.go

```diff
@@ -0,0 +1,550 @@
+// *** AUTO GENERATED CODE ***
+// @Product DDS
+// ---------------------------------------------------------------
+
+package dds
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+type backupError struct {
+	ErrorCode string `json:"error_code"`
+	ErrorMsg  string `json:"error_msg"`
+}
+
+func ResourceDdsBackup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDdsBackupCreate,
+		ReadContext:   resourceDdsBackupRead,
+		DeleteContext: resourceDdsBackupDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: backupImportState,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the ID of a DDS instance.`,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				Description: `Specifies the manual backup name.
+`,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`^[A-Za-z-_0-9]*$`),
+						"the value must be 4 to 64 characters in length and start with a letter"+
+							"(from A to Z or from a to z). It is case-sensitive and can contain only letters,"+
+							"digits (from 0 to 9), hyphens (-), and underscores (_)"),
+					validation.StringLenBetween(4, 64),
+				),
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				Description:  `Specifies the manual backup description`,
+				ValidateFunc: validation.StringLenBetween(0, 256),
+			},
+			"instance_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the name of a DDS instance.`,
+			},
+			"datastore": {
+				Type:        schema.TypeList,
+				Elem:        BackupDatastoreSchema(),
+				Computed:    true,
+				Description: `Indicates the database version.`,
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the backup type. `,
```

Comment:
```suggestion
				Description: `Indicates the backup type.`,
```
remove the extra space

Reply:
done